lib/workers/global/config/parse/file.ts

```diff
@@ -81,14 +81,7 @@ export async function getConfig(env: NodeJS.ProcessEnv): Promise<AllConfig> {
 
   await deleteNonDefaultConfig(env); // Try deletion only if RENOVATE_CONFIG_FILE is specified
 
-  const { isMigrated, migratedConfig } = migrateConfig(config);
-  if (isMigrated) {
-    logger.warn(
-      { originalConfig: config, migratedConfig },
-      'Config needs migrating',
-    );
-    config = migratedConfig;
-  }
+  config = await migrateAndValidateConfig(config, configFile);
   return config;
```

Comment:
```suggestion
  return migrateAndValidateConfig(config, configFile);
```
no need to wait here.